src/main/java/com/globallogic/amcr/persistence/dao/librarycomponent/BookDao.java

```diff
@@ -0,0 +1,78 @@
+package com.globallogic.amcr.persistence.dao.librarycomponent;
+
+import java.util.List;
+import java.util.UUID;
+
+import com.globallogic.amcr.persistence.dao.Dao;
+import com.globallogic.amcr.persistence.model.librarycomponent.Book;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Repository;
+
+import com.globallogic.amcr.persistence.mapper.librarycomponent.BookMapper;
+
+@Repository
+public class BookDao implements Dao<Book, Book> {
+    public static final Logger Log = LoggerFactory.getLogger(BookDao.class.getName());
+    final BookMapper bookMapper;
+
+    public BookDao(BookMapper bookMapper) {
+        this.bookMapper = bookMapper;
+    }
+
+    public Book save(Book book, UUID id) {
+        try {
+            book.setId(id);
+            bookMapper.save(book);
+            return book;
+        } catch (Exception e) {
+            throw new RuntimeException("DAO could not save book", e);
+        }
+    }
+
+    public Book get(UUID id) {
+        Log.trace("DAO requesting book with ID {}", id);
+        return bookMapper.get(id);
+    }
+
+    public List<Book> getAll() {
+        Log.trace("DAO requesting all books");
+        return bookMapper.getAll();
+    }
+
+    public Book reserve(UUID id, Book oldBook, Book reservedBook) {
+        reservedBook.setId(id);
+        if (oldBook.equals(reservedBook)) {
+            return reservedBook;
+        }
+        if (reservedBook.getTitle() == null) {
+            reservedBook.setTitle(oldBook.getTitle());
+        }
+        if (reservedBook.getAuthor() == null) {
+            reservedBook.setAuthor(oldBook.getAuthor());
+        }
+        if (reservedBook.getGenre() == null) {
+            reservedBook.setGenre(oldBook.getGenre());
+        }
+        if (reservedBook.getCover() == null) {
+            reservedBook.setCover(oldBook.getCover());
+        }
+        if (reservedBook.getTitle() == null) {
+            reservedBook.setTitle(oldBook.getTitle());
+        }
+        if (reservedBook.getTitle() == null) {
```

Comment:
This is duplicate of the above statement

Reply:
In fact this is the third copy of this if statement

Reply:
Just had to be sure it was the right title lol, fixed 👍